scenes/enemy-call-button/enemy-call-button.gd

```diff
@@ -0,0 +1,127 @@
+tool
+class_name EnemyCallButton
+extends Area2D
+
+signal enemy_start_timeout(enemy_call_button)
+
+onready var texture_progress = $Visible/TextureProgress
+onready var animation_player = $AnimationPlayer
+onready var popup_info = $PopupInfo
+onready var wave_info = $PopupInfo/MarginContainer/VBoxContainer/WaveInfo
+onready var collision_shape = $CollisionShape2D
+onready var texture_select = $Visible/TextureSelect
+onready var tween = $Tween
+
+export var id: int
+export(int, 0, 360) var direction: int setget on_dir_changed
+
+var project_size_y = ProjectSettings.get_setting("display/window/size/height")
+
+var wait_time: float
+var enemy_wave_description: String
+var call_button_selected: bool = false
+var mouse_inside_area: bool
+
+var can_process: bool = false
+
+func _ready():
+	var _ret = get_tree().connect("screen_resized", self, "_on_screen_resize")
+	visible = false
+
+
+func _process(delta):
+	if can_process:
```

Comment:
It's okay to use a boolean variable, however it's good to note that Godot has `set_process_unhandled_input`, `set_process` and other `set_process*` methods.

Reply:
Hi used the bool var because with the "tool" tag (to allow setting the direction in edit time) the script gives errors in the use of nodes references ...